src/AzureIoTHub.Portal/Server/Extensions/StringExtension.cs

```diff
@@ -34,5 +34,10 @@ public static string RemoveDiacritics(this string text)
 
             return regex.Replace(normalized, "-");
         }
+
+        public static string KeepAuthorizedCharacters(this string text)
+        {
+            return string.IsNullOrWhiteSpace(text) ? text : Regex.Replace(text, @"[^0-9a-zA-Z]+", "", RegexOptions.Compiled);
+        }
```

Comment:
Why not using https://github.com/CGI-FR/IoT-Hub-Portal/blob/c30edb91e69b9ab179cb67d4db2898bcd01e31cc/src/AzureIoTHub.Portal/Server/Extensions/StringExtension.cs#L23 instead of creating this fonction ?